### PR TITLE
Add caption normalization and roll-up layout mode

### PR DESCRIPTION
**Pull request by talk2MeGooseman**
## Summary
This PR enhances the closed captions system with automatic text normalization and introduces a new "roll-up" layout mode that displays each caption segment on its own line for improved readability.

### Caption Text Normalization
- Added `normalizeSegment()` helper that capitalizes the first character and appends terminal punctuation (period, question mark, exclamation mark, etc.) to caption segments
- Supports multiple scripts including Latin and CJK punctuation marks
- Handles edge cases like existing punctuation, quotes, and whitespace-only input
- Ensures sentence boundaries are visually clear when segments are joined together

### New Helper Functions
- `getCaptionQueue()` - Selects the appropriate caption queue (default or translation) for the active viewer language
- `assembleCaptionText()` - Assembles normalized caption segments into a single flowing text string
- `assembleCaptionLines()` - Assembles normalized segments as individual lines with stable IDs for roll-up layout
- `isCaptionsHidden()` - Determines caption box visibility based on viewer toggle and content availability

### Roll-Up Layout Mode
- Added `rollUpCaptions` setting (defaults to `true`) that controls caption display format
- When enabled: each recognized segment displays on its own line, making sentence boundaries explicit
- When disabled: segments flow together in a single paragraph (legacy behavior)
- Implemented in both desktop (`ClosedCaption.jsx`) and mobile (`MobileClosedCaption.jsx`) components
- Added `RollUpCaptionsOptionButton` menu item to toggle the layout mode

### Styling Updates
- Extended `CaptionText` styled component with `$block` prop to support line-by-line rendering
- Maintains proper spacing and formatting for both layout modes

### Test Coverage
- Comprehensive test suite for all new helper functions covering edge cases
- Updated existing component tests to expect normalized text (with capitalization and punctuation)
- Added layout mode tests for both desktop and mobile components
- Tests verify correct behavior for translations and interim text handling

## Implementation Details
- Interim text only renders in default language mode to prevent empty boxes in translation mode
- Caption normalization is applied consistently across all display modes and languages
- The roll-up setting is persisted in Redux state and can be toggled via the display settings menu
- All changes maintain backward compatibility with existing caption functionality

https://claude.ai/code/session_011454WHEWgCDzY8UjBHBax1

### CODE REVIEW

File: src/components/Captions/MobileClosedCaption.jsx [hunk 1]
```diff
@@ -36,19 +39,19 @@ function MobileClosedCaption() {
...
-      translations,
-    )
-      .map(({ text }) => text)
-      .join(' ')
-  }
+  // Roll-up (line-per-sentence) is the default; viewers can opt back into the
+  // single flowing paragraph via the display settings menu.
+  const rollUpCaptions = configSettings.rollUpCaptions ?? true
+  const captionLines = assembleCaptionLines(
+    configSettings.viewerLanguage,
```
[the remark this answers is not in the document]
Comment (talk2MeGooseman) [reply src/components/Captions/MobileClosedCaption.jsx:46]: Thanks — the empty-overlay observation is valid, but we're intentionally leaving the mobile panel's visibility as-is in this PR (it's pre-existing behavior; `MobileClosedCaption` never had hide logic).

The `hideCC` half of the suggestion would actually regress mobile: `hideCC` defaults to `true`, and unlike the video overlay the mobile panel has no `VisibilityToggle` (it returns `null` when `!isVideoOverlay()`), while `Twitch.jsx` only stops the captions subscription for `hideCC` on the overlay. So wiring `isCaptionsHidden(hideCC, …)` into the mobile container would hide captions by default with no way for a mobile viewer to turn them back on.

The "empty content → dark full-screen panel" case is real but separate from `hideCC`; we'd rather address it on its own (hide only when there's no final/interim content) as a follow-up rather than couple it to `hideCC` here.

---
_Generated by [Claude Code](https://claude.ai/code)_
